components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -1483,15 +1494,14 @@ export function createWalletApi (
             }
           }
         },
-        providesTags: (res, err, arg) =>
+        providesTags: (balancesResult, err, arg) =>
           err
-            ? ['UNKNOWN_ERROR']
-            : [
-                {
+            ? ['TokenBalancesForChainId', 'UNKNOWN_ERROR']
+            : (balancesResult &&
+                Object.keys(balancesResult).map((tokenAddress) => ({
                   type: 'TokenBalancesForChainId',
-                  id: `${arg.address}-${arg.coin}-${arg.chainId}`
-                }
-              ]
+                  id: `${arg.address}-${arg.coin}-${arg.chainId}-${tokenAddress}`
+                }))) || ['TokenBalancesForChainId']
```

Comment:
Then refactor here based on the other suggestion.

Reply:
refactored